integration/spark/src/main/scala/org/apache/spark/sql/events/MergeIndexEventListener.scala

```diff
@@ -140,19 +141,47 @@ class MergeIndexEventListener extends OperationEventListener with Logging {
                   .get
                   .filterNot(streamingSegment.contains(_))
               }
+            validSegments.foreach { segment =>
+              if (segmentsToMerge.contains(segment.getSegmentNo)) {
+                val segmentFile = segment.getSegmentFileName
+                val sfs = new SegmentFileStore(carbonMainTable.getTablePath, segmentFile)
```

Comment:
looks like, segment file is read twice. Please check and handle

Reply:
Done.